operator/v1alpha1/types_etcdbackup.go

```diff
@@ -0,0 +1,102 @@
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type EtcdBackup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec EtcdBackupSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status EtcdBackupStatus `json:"status"`
+}
+
+type EtcdBackupSpec struct {
+	// PVCName specifies the name of the PersistentVolumeClaim which binds a PersistentVolume where the
+	// etcd backup file would be saved
+	// +kubebuilder:validation:Required
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="pvcName is immutable once set"
+	PVCName string `json:"pvcName"`
+}
+
+const (
+	// Etcd backup is running
+	Running EtcdBackupState = "Running"
+	// Etcd backup is completed
+	Completed EtcdBackupState = "Completed"
+	// Etcd backup failed
+	Failed EtcdBackupState = "Failed"
+	// Etcd backup is pending
+	Pending EtcdBackupState = "Pending"
+)
+
+// etcdBackupState declares valid gathering state types
+// +kubebuilder:validation:Optional
+// +kubebuilder:validation:Enum=Running;Completed;Failed;Pending
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Running' && self == 'Pending')", message="etcdBackupState cannot transition from Running to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Pending')", message="etcdBackupState cannot transition from Completed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Pending')", message="etcdBackupState cannot transition from Failed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Running')", message="etcdBackupState cannot transition from Completed to Running"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Running')", message="etcdBackupState cannot transition from Failed to Running"
+type EtcdBackupState string
+
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.startTime) || has(self.startTime))",message="cannot remove startTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.finishTime) || has(self.finishTime))",message="cannot remove finishTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.etcdBackupState) || has(self.etcdBackupState))",message="cannot remove etcdBackupState attribute from status"
+// +kubebuilder:validation:Optional
+type EtcdBackupStatus struct {
```

Comment:
Do you need a link to the pod or job running this backup?

Reply:
in the current implementation I added a "back pointer" label on the job to the backup CRD name. The other direction is a status condition. Do you think it makes sense to have this as a typed field?

Reply:
@tjungblu For the "back pointer" from the Job=>Backup CRD we can set the `metadata.ownerReferences` field so that the Backup CRD is identified as the parent on the Job's metadata.

For the Backup CRD=>Job direction, I was also thinking that the Job name would be visible in the status condition once it is created or pending.
But @deads2k let us know if you think a typed status field would be better.

Reply:
great idea, will update the other controller accordingly

Reply:
I can add a typed status field to reference the Job associated with the EtcdBackup. Name or better yet some kind of object reference, like the owner ref has it.